src/ray/gcs/gcs_server/gcs_server.cc

```diff
@@ -598,7 +598,10 @@ void GcsServer::InitMonitorServer() {
       gcs_node_manager_,
       cluster_resource_scheduler_->GetClusterResourceManager(),
       gcs_resource_manager_,
-      gcs_placement_group_manager_);
+      gcs_placement_group_manager_,
+      // This value comes from derefenceing a unique_ptr, so it could be unsafe
+      // to store. Should we switch to shared_ptrs and pass that instead?
+      kv_manager_->GetInstance());
```

Comment:
I think it's OK. Ideally, no shared ptr should be passed around and all things are owned by GcsServer. With this, it actually should make things clearer.